regression/Makefile

```diff
@@ -20,18 +20,51 @@ DIRS = ansi-c \
        test-script \
        # Empty last line
 
-# Check for the existence of $dir. Tests under goto-gcc cannot be run on
-# Windows, so appveyor.yml unlinks the entire directory under Windows.
+# Run all test directories in sequence
 test:
 	@for dir in $(DIRS); do \
-	  if [ -d "$$dir" ]; then \
-	    $(MAKE) -C "$$dir" test || exit 1; \
-	  fi; \
+        $(MAKE) "test-$$dir" || exit 1; \
 	done;
 
+# Pattern to execute a single directory, checking for existence of the directory
+# first. Tests under goto-gcc cannot be run on Windows, so appveyor.yml unlinks
+# the entire directory under Windows.
+test-%:
+	@if [ -d "$*" ]; then \
+		$(MAKE) -C "$*" test || exit 1; \
+	fi;
+
+
+# Run all test directories in batch mode, enabling parallel execution of
+# test directories while avoiding interleaving of output, at the expense
+# of less interactive feedback.
+BATCH_TARGETS = $(patsubst %,.%-batch-output, $(DIRS))
+
+#.PHONY: test-batch $(BATCH_TARGETS)
+
+test-batch: $(BATCH_TARGETS)
+	@for batch in "$<" ; do \
+		if [ -f "$$batch" ]; then \
+			cat "$$batch"; \
+			rm "$$batch"; \
+		fi; \
+	done;
+
+# Run a single test directory, capturing it's stdout/stderr to
+# a file
+.%-batch-output:
```

Comment:
Can/are equivalent changes done for the CMake version?

Reply:
The CMake version uses CTest to run the tests, at that already knows how to do this (and does it more elegantly, to be honest). This is just makeing (no pun intended) the Makefile builds catchup.